packages/framework/src/Model/Product/Search/Export/ProductSearchExportConsumer.php

```diff
@@ -0,0 +1,55 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Shopsys\FrameworkBundle\Model\Product\Search\Export;
+
+use Doctrine\ORM\EntityManagerInterface;
+use OldSound\RabbitMqBundle\RabbitMq\ConsumerInterface;
+use PhpAmqpLib\Message\AMQPMessage;
+
+class ProductSearchExportConsumer implements ProductSearchExportConsumerInterface, ConsumerInterface
+{
+    /**
+     * @var \Doctrine\ORM\EntityManagerInterface
+     */
+    protected $entityManager;
+
+    /**
+     * @var \Shopsys\FrameworkBundle\Model\Product\Search\Export\ProductSearchExportFacade
+     */
+    protected $productSearchExportFacade;
+
+    /**
+     * @param \Doctrine\ORM\EntityManagerInterface $entityManager
+     * @param \Shopsys\FrameworkBundle\Model\Product\Search\Export\ProductSearchExportFacade $productSearchExportFacade
+     */
+    public function __construct(
+        EntityManagerInterface $entityManager,
+        ProductSearchExportFacade $productSearchExportFacade
+    ) {
+        $this->entityManager = $entityManager;
+        $this->productSearchExportFacade = $productSearchExportFacade;
+    }
+
+    /**
+     * @param \PhpAmqpLib\Message\AMQPMessage $message
+     * @return int
+     */
+    public function execute(AMQPMessage $message): int
```

Comment:
Have you ever looked at https://github.com/VasekPurchart/Rabbit-Mq-Consumer-Handler-Bundle? Maybe it can help with something.